lang/src/org/partiql/lang/eval/LikeMatchingAutomata.kt

```diff
@@ -165,18 +182,19 @@ private open class DFAState(val nfaStates: MutableSet<NFAState>,
  * - [isStartState] true when this State is a Start state, false othewise
  * - [outgoing] map of alphabet letter to NFA State
  */
-private class NFAState(val stateNumber: Int,
-               val isAccepting: Boolean,
-               val isStartState: Boolean,
-               val outgoing: MutableMap<Alphabet, MutableSet<NFAState>> = mutableMapOf<Alphabet, MutableSet<NFAState>>()) {
+private class NFAState(
+    val stateNumber: Int,
+    val isAccepting: Boolean,
+    val isStartState: Boolean,
+    val outgoing: MutableMap<Alphabet, MutableSet<NFAState>> = HashMap()) {
 
     fun get(transition: Alphabet): Set<NFAState> =
-        outgoing[transition]?.let { it } ?: setOf<NFAState>()
+        outgoing[transition] ?: hashSetOf()
```

Comment:
We're using *unordered* sets which are `java.util.HashSet` instead of `java.util.LinkedHashSet` which should reduce memory and CPU overhead a little.

Reply:
Can there be a use-case where the order of elements is important? Also, as far as I know, the `LinkedHashSet` is almost as fast and efficient as `HashSet` asymptotically speaking.

Reply:
Well, all of the tests pass and functionality seems to be unaffected.  I don't think order is important.